pandas/core/series.py

```diff
@@ -2696,28 +2696,55 @@ def reindex_axis(self, labels, axis=0, **kwargs):
         return self.reindex(index=labels, **kwargs)
 
     def memory_usage(self, index=True, deep=False):
-        """Memory usage of the Series
+        """
+        Return the memory usage of the Series.
+
+        The memory usage can optionally include the contribution of
+        the index and of elements of `object` dtype.
 
         Parameters
         ----------
-        index : bool
-            Specifies whether to include memory usage of Series index
-        deep : bool
-            Introspect the data deeply, interrogate
-            `object` dtypes for system-level memory consumption
+        index : bool, default True
+            Specifies whether to include the memory usage of the Series index.
+        deep : bool, default False
+            If True, introspect the data deeply by interrogating
+            `object` dtypes for system-level memory consumption, and include
+            it in the returned value.
 
         Returns
         -------
-        scalar bytes of memory consumed
-
-        Notes
-        -----
-        Memory usage does not include memory consumed by elements that
-        are not components of the array if deep=False
+        int
+            Bytes of memory consumed.
 
         See Also
         --------
-        numpy.ndarray.nbytes
+        numpy.ndarray.nbytes : Total bytes consumed by the elements of the
+            array.
+
+        Examples
+        --------
+
+        >>> s = pd.Series(range(3))
+        >>> s.memory_usage()
+        104
+
+        Not including the index gives the size of the rest of the data, which
+        is necessarily smaller:
+
+        >>> s.memory_usage(index=False)
+        24
+
+        The memory footprint of `object` values is ignored by default:
+
+        >>> class MyClass: pass
+        >>> s = pd.Series(MyClass())
```

Comment:
I would maybe use an existing python object (something from the standard library, eg decimal.Decimal, IPAddress object, ... or your other favorite python object)

Reply:
Or, maybe also useful, to just use some strings. Because of storing them as object (typical gotcha in pandas), there memory is not reflected well by default in this

Reply:
Yes, I like the idea of using strings.

You'll need to make sure the string is is not interned by the interprepter, so don't use just `pd.Series(['a', 'b'])`

Reply:
Why not use interned strings? They are longer than their pointer, so they also are a good example:
```
>>> s = pd.Series(["a", "b"])
>>> s.memory_usage()
96
>>> s.memory_usage(deep=True)
212
```

Reply:
so apparantly it is giving the memory of that object, so for the example that looks fine (but I would expect that if you have repeated interned strings, it might then not give the correct result)

Reply:
A repeated interned string does not give the real memory footprint indeed:
```
>>> id("a")
4397773352
>>> id("a")
4397773352
>>> s = pd.Series(["a"]*1000)
>>> s.memory_usage()
8080
>>> s.memory_usage(deep=True)
66080
```